sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -456,6 +457,22 @@ static struct snd_soc_ops rt1308_i2s_ops = {
 	.hw_params = rt1308_i2s_hw_params,
 };
 
+/* these wrappers are only needed to avoid typecast compilation errors */
+static int sdw_startup(struct snd_pcm_substream *substream)
+{
+	return sdw_startup_stream(substream);
+}
+
+static void sdw_shutdown(struct snd_pcm_substream *substream)
+{
+	sdw_shutdown_stream(substream);
+}
+
+static const struct snd_soc_ops sdw_ops = {
+	.startup = sdw_startup,
+	.shutdown = sdw_shutdown,
+};
+
```

Comment:
You may want to add below in this commit
```patch
--- a/sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c
+++ b/sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c
@@ -776,7 +776,7 @@ static int create_sdw_codec_dai(struct device *dev,
 			return -ENOMEM;
 
 		init_dai_link(dai_links, *id, name, 1 - i, i, cpus, cpu_name,
-			      codec, link->num_adr, *id, NULL, NULL);
+			      codec, link->num_adr, *id, NULL, &sdw_ops);
 		set_codec_init_func(link, dai_links + *id, 1 - i);
 		(*id)++;
 		j++;
```

Reply:
thanks, I will check it.

Reply:
Yes this works, thanks @bardliao